ddtrace/tracer/spancontext.go

```diff
@@ -242,6 +242,9 @@ func (c *spanContext) baggageItem(key string) string {
 }
 
 func (c *spanContext) meta(key string) (val string, ok bool) {
+	if c.span == nil {
+		return "", false
+	}
```

Comment:
@rarguelloF I wonder if we could fix this by allowing to fail if `c.span` is nil, but checking that the span isn't nil in the code that relies on `meta` in `sqlcomment.go`:

```go
	case DBMPropagationModeService:
		if ctx, ok := spanCtx.(*spanContext); ok && ctx.span != nil {
			if e, ok := ctx.meta(ext.Environment); ok && e != "" {
				tags[sqlCommentEnv] = e
			}
```

My rationale is that introducing this check here may yield to hidden behaviour. Also, to me this seems an example of not enough coverage, as it only impact a single case of DBM propagation.

We should add a test case covering this case, if possible. WDYT?

Reply:
> introducing this check here may yield to hidden behaviour.

I don't think any caller of a non-nil `*spanContext` of `.meta()` should be expecting a panic in general + the only callers of this function are in this `sqlcomment.go` file.

Making this check would make things a little bit more redundant (we would have to check this in N places where this method is called) and expects the caller to know these internal implementation details (the caller needs to know that a non-nil `*spanContext` might have a nil `span` - which btw is a private field), which IMHO is not a good thing...

Reply:
> I don't think any caller of a non-nil *spanContext of .meta() should be expecting a panic in general + the only callers of this function are in this sqlcomment.go file.

That's why I suggested doing this the other way around. We are adding code that has general impact for somethingt that only happens in a corner case of a module in the whole tracer 😄 Also, doesn't seem strange that only impacts this module but not the rest?

These are internal APIs, so I'm fine having extra requirements.

Reply:
> Agree! Isn't the new test case that I added enough? or were you thinking on something else/different?

I had in mind to actually test the switch branch that panics when you run `Inject` on a recently extracted context.

Reply:
As discussed in private:
- Will move the nil check to the callers instead.
- Remove the `spanContext.meta` method to `getMeta` function